homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""Coordinator for handling data fetching and updates."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+from madvr.madvr import Madvr
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from . import MadVRConfigEntry
+
+
+class MadVRCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Madvr coordinator for Envy (push-based API)."""
+
+    config_entry: MadVRConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: Madvr,
+    ) -> None:
+        """Initialize madvr coordinator."""
+        super().__init__(hass, _LOGGER, name=DOMAIN)
+        self.mac = self.config_entry.unique_id
+        assert self.mac
```

Comment:
```suggestion
        assert self.config_entry.unique_id
        self.mac = self.config_entry.unique_id
```
This way `self.mac` is always `str` so we can avoid the `cast`s in `remote.py`